tests/unit/controllers/api/v1/test_actions_rbac.py

```diff
@@ -165,6 +273,147 @@ def test_get_all_limit_larget_than_page_size(self):
+    @mock.patch("st2api.controllers.v1.actions.clone_action_files")
+    @mock.patch.object(action_validator,
+                       "validate_action",
+                       mock.MagicMock(return_value=True))
+    def test_clone_action_success(self, mock_clone_action):
+        user_db = self.users["action_clone"]
+        self.use_user(user_db)
+        self.__do_post(ACTION_2)
+        dest_data_body = {
+            "dest_pack": ACTION_3["pack"],
+            "dest_action": "clone_action_2",
+        }
+        source_ref_or_id = "%s.%s" % (ACTION_2["pack"], ACTION_2["name"])
+        clone_resp = self.__do_clone(dest_data_body, source_ref_or_id)
+        self.assertEqual(clone_resp.status_code, http_client.CREATED)
+
+    @mock.patch.object(os.path, "isdir", mock.MagicMock(return_value=True))
+    @mock.patch("st2api.controllers.v1.actions.remove_temp_action_files")
+    @mock.patch("st2api.controllers.v1.actions.restore_temp_action_files")
+    @mock.patch("st2api.controllers.v1.actions.temp_backup_action_files")
+    @mock.patch("st2api.controllers.v1.actions.clone_action_files")
+    @mock.patch.object(action_validator,
+                       "validate_action",
+                       mock.MagicMock(return_value=True))
+    def test_clone_overwrite_action_success(self,
+                                            mock_clone_action,
+                                            mock_backup_files,
+                                            mock_restore_files,
+                                            mock_remove_backup):
+        user_db = self.users["action_clone"]
+        self.use_user(user_db)
+        self.__do_post(ACTION_2)
+        self.__do_post(ACTION_3)
+        dest_data_body = {
+            "dest_pack": ACTION_3["pack"],
+            "dest_action": ACTION_3["name"],
+            "overwrite": True,
+        }
+        source_ref_or_id = "%s.%s" % (ACTION_2["pack"], ACTION_2["name"])
+        clone_resp = self.__do_clone(dest_data_body, source_ref_or_id)
+        self.assertEqual(clone_resp.status_code, http_client.CREATED)
+
+    @mock.patch.object(action_validator,
+                       "validate_action",
+                       mock.MagicMock(return_value=True))
+    def test_clone_action_no_source_action_view_permission(self):
+        user_db = self.users["action_create"]
+        self.use_user(user_db)
+        post_resp = self.__do_post(ACTION_2)
+        user_db = self.users["no_permissions"]
+        self.use_user(user_db)
+        dest_data_body = {
+            "dest_pack": ACTION_3["pack"],
+            "dest_action": "clone_action_3",
+        }
+        source_ref_or_id = "%s.%s" % (ACTION_2["pack"], ACTION_2["name"])
+        clone_resp = self.__do_clone(dest_data_body,
+                                     source_ref_or_id,
+                                     expect_errors=True)
+        expected_msg = ('User "no_permissions" doesn\'t have required permission '
```

Comment:
Please use string formatting to pass in user name instead of hard coding "no_permissions".